Allow TRUNCATE on base tables

yugo-n opened the pull request:
If we don't have any outer-joins, the view can be truncated when
a base table is truncated. Otherwise, if we have any outer-joins,
a simple refresh is performed.

yugo-n committed:
Allow TRUNCATE on base tables

If we don't have any outer-joins, the view can be truncated when
a base table is truncated. Otherwise, if we have any outer-joins,
a simple refresh is performed.

thoshiai reviewed (changes requested):
I think this code is  good. However, regression test and documentation
have not been fixed yet. I would like you to add these.

yugo-n committed:
Do not truncate views which contain aggregates without GROUP

Aggregate views without a GROUP clause always have one row. Therefore,
if a base table is truncated, the view will not be empty and will contain
a row with NULL value (or 0 for count()). So, in this case, we refresh the
view instead of truncating it.

Also, tests are added

yugo-n commented:
I added tests.

Also, I fixed the behavior when the view has aggregates without GROUP clause.

Aggregate views without a GROUP clause always have one row. Therefore,
if a base table is truncated, the view will not be empty and will contain
a row with NULL value (or 0 for count()). So, in this case, we refresh the
view instead of truncating it.
